presto-main/src/main/java/io/prestosql/execution/DynamicFilterConfig.java

```diff
@@ -0,0 +1,244 @@
+        "dynamic-filtering-max-per-driver-size",
+        "experimental.dynamic-filtering-max-per-driver-size",
+        "dynamic-filtering-range-row-limit-per-driver"
+})
+public class DynamicFilterConfig
+{
+    private boolean enableDynamicFiltering = true;
+    private boolean enableLargeDynamicFilters;
+    private Duration dynamicFilteringRefreshInterval = new Duration(200, MILLISECONDS);
+
+    private int smallBroadcastMaxDistinctValuesPerDriver = 100;
+    private DataSize smallBroadcastMaxSizePerDriver = DataSize.of(10, KILOBYTE);
+    private int smallBroadcastRangeRowLimitPerDriver;
+    private int smallPartitionedMaxDistinctValuesPerDriver = 100;
+    private DataSize smallPartitionedMaxSizePerDriver = DataSize.of(10, KILOBYTE);
+    private int smallPartitionedRangeRowLimitPerDriver;
+
+    private int largeBroadcastMaxDistinctValuesPerDriver = 5_000;
+    private DataSize largeBroadcastMaxSizePerDriver = DataSize.of(100, KILOBYTE);
+    private int largeBroadcastRangeRowLimitPerDriver = 50_000;
+    private int largePartitionedMaxDistinctValuesPerDriver = 500;
+    private DataSize largePartitionedMaxSizePerDriver = DataSize.of(10, KILOBYTE);
+    private int largePartitionedRangeRowLimitPerDriver = 5_000;
+
+    public boolean isEnableDynamicFiltering()
+    {
+        return enableDynamicFiltering;
+    }
+
+    @Config("enable-dynamic-filtering")
+    @LegacyConfig("experimental.enable-dynamic-filtering")
+    public DynamicFilterConfig setEnableDynamicFiltering(boolean enableDynamicFiltering)
+    {
+        this.enableDynamicFiltering = enableDynamicFiltering;
+        return this;
+    }
+
+    public boolean isEnableLargeDynamicFilters()
+    {
+        return enableLargeDynamicFilters;
+    }
+
+    @Config("enable-large-dynamic-filters")
+    public DynamicFilterConfig setEnableLargeDynamicFilters(boolean enableLargeDynamicFilters)
+    {
+        this.enableLargeDynamicFilters = enableLargeDynamicFilters;
+        return this;
+    }
+
+    @MinDuration("1ms")
+    @MaxDuration("10s")
+    @NotNull
+    public Duration getDynamicFilteringRefreshInterval()
+    {
+        return dynamicFilteringRefreshInterval;
+    }
+
+    @Config("experimental.dynamic-filtering-refresh-interval")
+    public DynamicFilterConfig setDynamicFilteringRefreshInterval(Duration dynamicFilteringRefreshInterval)
+    {
+        this.dynamicFilteringRefreshInterval = dynamicFilteringRefreshInterval;
+        return this;
+    }
+
+    public int getSmallBroadcastMaxDistinctValuesPerDriver()
+    {
+        return smallBroadcastMaxDistinctValuesPerDriver;
+    }
+
+    @Config("dynamic-filtering.small-broadcast.max-distinct-values-per-driver")
+    public DynamicFilterConfig setSmallBroadcastMaxDistinctValuesPerDriver(int smallBroadcastMaxDistinctValuesPerDriver)
+    {
+        this.smallBroadcastMaxDistinctValuesPerDriver = smallBroadcastMaxDistinctValuesPerDriver;
+        return this;
+    }
+
+    @MaxDataSize("1MB")
```

Comment:
make it 10MB

Reply:
updated max annotation for large-partitioned.max-size-per-driver to 5MB, large-broadcast.max-size-per-driver to 50MB and kept it capped at 1MB for the small DF cases.